session.go

```diff
@@ -31,3 +31,30 @@ type Session interface {
 	// session implementation.
 	Version() (msize int, version string)
 }
+
+// WriteFull writes the whole p slice to the specified fid
+func WriteFull(s Session, ctx context.Context, fid Fid, p []byte, offset int64) error {
```

Comment:
Place these in a separate file. They are just methods to use with session. See `readdir.go` for an example.